realm-jni/src/util.cpp

```diff
@@ -117,9 +112,9 @@ void ThrowException(JNIEnv* env, ExceptionKind exception, const std::string& cla
             message = classStr + " " + itemStr;
             break;
 
-        case Unspecified:
-            jExceptionClass = env->FindClass("java/lang/RuntimeException");
```

Comment:
I assume this is only temporary until core manages to make their thrown exceptions better, or is it assumed that going forward all recoverable error cases will result in a named exception we can handle specifically? And we then have to add those as they are implemented, e.g. QueryException?